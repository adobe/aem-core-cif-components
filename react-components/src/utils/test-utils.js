/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../__mocks__/i18nForTests';
import ConfigContextProvider from '../context/ConfigContext';
import UserContextProvider from '../context/UserContext';

import mutationChangePassword from './mocks/mutationChangePassword';
import mutationCreateCustomer from './mocks/mutationCreateCustomer';
import mutationCreateDuplicateCustomer from './mocks/mutationCreateCustomerDuplicate';
import mutationDeleteCustomerAddress from './mocks/mutationDeleteCustomerAddress';
import mutationGenerateToken from './mocks/mutationGenerateToken';
import mutationGenerateTokenWrongPassword from './mocks/mutationGenerateTokenWrongPassword';
import mutationMergeCarts from './mocks/mutationMergeCarts';
import mutationPlaceOrder from './mocks/mutationPlaceOrder';
import mutationRevokeToken from './mocks/mutationRevokeToken';
import mutationShippingAddress from './mocks/mutationShippingAddress';
import queryCart from './mocks/queryCart';
import queryCountries from './mocks/queryCountries';
import queryCustomerCart from './mocks/queryCustomerCart';
import queryCustomerDetails from './mocks/queryCustomerDetails';
import queryCustomerInformation from './mocks/queryCustomerInformation';
import queryEmptyCart from './mocks/queryEmptyCart';
import queryNewCart from './mocks/queryNewCart';

const defaultMocks = [
    mutationChangePassword,
    mutationCreateCustomer,
    mutationCreateDuplicateCustomer,
    mutationDeleteCustomerAddress,
    mutationGenerateToken,
    mutationGenerateTokenWrongPassword,
    mutationMergeCarts,
    mutationPlaceOrder,
    mutationRevokeToken,
    mutationShippingAddress,
    queryCart,
    queryCountries,
    queryCustomerCart,
    queryCustomerDetails,
    queryCustomerInformation,
    queryEmptyCart,
    queryNewCart
];

const defaultConfig = {
    storeView: 'default',
    graphqlEndpoint: 'none'
};

// eslint-disable-next-line react/display-name
const allProviders = (config, userContext, mocks) => ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            <ConfigContextProvider config={config || defaultConfig}>
                <UserContextProvider initialState={userContext}>
                    <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
                </UserContextProvider>
            </ConfigContextProvider>
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options = {}) => {
    const { config, userContext, mocks = defaultMocks, ...renderOptions } = options;
    return render(ui, { wrapper: allProviders(config, userContext, mocks), ...renderOptions });
};

export * from '@testing-library/react';
export { customRender as render };
