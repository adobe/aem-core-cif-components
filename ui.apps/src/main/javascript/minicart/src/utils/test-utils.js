/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import QUERY_CART_DETAILS from '../queries/query_cart_details.graphql';

const emptyCartId = 'empty';
const mockCartId = '123ABC';

const mocks = [
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: emptyCartId
            }
        },
        result: {
            data: {
                cart: {
                    email: null,
                    shipping_addresses: [],
                    prices: {
                        grand_total: {
                            currency: 'USD',
                            value: 0
                        }
                    },
                    selected_payment_method: {
                        code: '',
                        title: ''
                    },
                    billing_address: {
                        city: null,
                        country: {
                            code: null
                        },
                        lastname: null,
                        firstname: null,
                        region: {
                            code: null
                        },
                        street: [''],
                        postcode: null,
                        telephone: null
                    },
                    available_payment_methods: [
                        {
                            code: 'cashondelivery',
                            title: 'Cash On Delivery'
                        },
                        {
                            code: 'banktransfer',
                            title: 'Bank Transfer Payment'
                        },
                        {
                            code: 'checkmo',
                            title: 'Check / Money order'
                        },
                        {
                            code: 'free',
                            title: 'No Payment Information Required'
                        }
                    ],
                    items: []
                }
            }
        }
    },
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: mockCartId
            }
        },
        result: {
            data: {
                prices: {
                    grand_total: {
                        currency: 'USD',
                        value: 0
                    }
                },
                items: []
            }
        }
    }
];

const AllProviders = ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            {children}
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
