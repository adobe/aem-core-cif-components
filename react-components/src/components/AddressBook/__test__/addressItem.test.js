/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { fireEvent } from '@testing-library/react';
import { render } from 'test-utils';
import UserContextProvider from '../../../context/UserContext';
import AddressItem from '../addressItem';

describe('<AddressItem>', () => {
    const mockAddress = {
        id: 'my-address-id',
        region: {
            region_code: 'LA'
        },
        street: ['14 Stamford Court'],
        default_shipping: false,
        default_billing: false
    };

    it('renders the component', () => {
        const { asFragment } = render(<AddressItem address={mockAddress} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with list display type', () => {
        const { asFragment } = render(<AddressItem address={mockAddress} displayType={'list'} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component without default address checkbox', () => {
        const mockDefaultAddress = {
            ...mockAddress,
            default_shipping: true,
            default_billing: true
        };
        const { asFragment } = render(<AddressItem address={mockDefaultAddress} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with delete address modal', () => {
        const { asFragment } = render(<AddressItem address={mockAddress} />, {
            userContext: { deleteAddress: { id: 'my-address-id' } }
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('handle click event of edit button', () => {
        const handler = jest.fn(state => state);

        const { getByText } = render(
            <UserContextProvider reducerFactory={() => handler}>
                <AddressItem address={mockAddress} />
            </UserContextProvider>
        );
        fireEvent.click(getByText('Edit'));
        expect(handler.mock.calls.length).toEqual(1);
    });

    it('handle click event of delete button', () => {
        const handler = jest.fn(state => state);

        const { getByText } = render(
            <UserContextProvider reducerFactory={() => handler}>
                <AddressItem address={mockAddress} />
            </UserContextProvider>
        );
        fireEvent.click(getByText('Delete'));
        expect(handler.mock.calls.length).toEqual(1);
    });
});
