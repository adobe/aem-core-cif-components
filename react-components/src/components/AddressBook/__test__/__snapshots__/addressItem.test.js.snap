// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressItem> renders the component 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="summary"
    >
      <div
        class="name"
      >
        <strong>
           
        </strong>
      </div>
      <div
        class="street"
      >
        14 Stamford Court
      </div>
      <div
        class="cityRegion"
      >
        , LA
      </div>
      <div
        class="country"
      />
    </div>
    <div
      class="actions"
    >
      <button
        class="editButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Edit
        </span>
      </button>
      <button
        class="deleteButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AddressItem> renders the component with delete address modal 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="summary"
    >
      <div
        class="name"
      >
        <strong>
           
        </strong>
      </div>
      <div
        class="street"
      >
        14 Stamford Court
      </div>
      <div
        class="cityRegion"
      >
        , LA
      </div>
      <div
        class="country"
      />
    </div>
    <div
      class="actions"
    >
      <button
        class="editButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Edit
        </span>
      </button>
      <button
        class="deleteButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Delete
        </span>
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="cancel"
      >
        <button
          class="root_normalPriority"
          type="button"
        >
          <span
            class="content"
          >
            Cancel
          </span>
        </button>
      </div>
      <div
        class="delete"
      >
        <button
          class="root_highPriority"
          type="button"
        >
          <span
            class="content"
          >
            Delete
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AddressItem> renders the component with list display type 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <div
      class="summary"
    >
      <div
        class="name"
      >
        <strong>
           
        </strong>
      </div>
      <div
        class="street"
      >
        14 Stamford Court
      </div>
      <div
        class="cityRegion"
      >
        , LA
      </div>
      <div
        class="country"
      />
    </div>
    <div
      class="actions"
    >
      <button
        class="editButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Edit
        </span>
      </button>
      <button
        class="deleteButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="3 6 5 6 21 6"
            />
            <path
              d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AddressItem> renders the component without default address checkbox 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="summary"
    >
      <div
        class="name"
      >
        <strong>
           
        </strong>
      </div>
      <div
        class="street"
      >
        14 Stamford Court
      </div>
      <div
        class="cityRegion"
      >
        , LA
      </div>
      <div
        class="country"
      />
    </div>
    <div
      class="actions"
    >
      <button
        class="editButton"
      >
        <span
          class="icon"
        >
          <svg
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
            />
            <path
              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
            />
          </svg>
        </span>
        <span
          class="label"
        >
          Edit
        </span>
      </button>
      <div
        class="defaultTag"
      >
        <span>
          Default
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
